NBitcoin.Tests/script_tests.cs

```diff
@@ -401,13 +401,17 @@ private void EnsureHasLibConsensus()
 			{
 				libConsensusDll = "libbitcoinconsensus-0.dll";
 			}
+			else if(RuntimeInformation.IsOSPlatform(OSPlatform.OSX))
+			{
+				libConsensusDll = "libbitcoinconsensus.0.dylib";
+			}
 			else if(RuntimeInformation.IsOSPlatform(OSPlatform.Linux))
 			{
 				libConsensusDll = "libbitcoinconsensus.so";
 			}
 			else
 			{
-				libConsensusDll = "libbitcoinconsensus.0.dylib";
+				throw new NotSupportedException("Unknown operating system");
 			}

```

Comment:
it seems we don't need all to have into account the `lib{whatever}.{dll|so|dylib}` because .NET can deal with that automatically. see: https://stackoverflow.com/questions/44745613/create-c-c-library-in-xcode-and-import-it-in-c-sharp-netcore

Reply:
Only thing I know is that it is a mess. .NET Framework might handle differently from .NET Core.